extension/content/components/recipes/form/arguments/MessagingExperiment.js

```diff
@@ -0,0 +1,221 @@
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  Col,
+  ControlLabel,
+  Divider,
+  FormGroup,
+  HelpBlock,
+  Icon,
+  IconButton,
+  Input,
+  InputNumber,
+  Panel,
+  Row,
+  TagPicker,
+} from "rsuite";
+
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+import ToggleField from "devtools/components/recipes/form/arguments/fields/ToggleField";
+import InputField from "devtools/components/recipes/form/arguments/fields/InputField";
+import JsonEditor from "devtools/components/common/JsonEditor";
+
+export default function MessagingExperiment() {
+  return (
+    <>
+      <InputField label="Experiment Slug" name="slug" />
+
+      <ToggleField label="Prevent New Enrollment" name="isEnrollmentPaused">
+        Prevents new users from joining the experiment cohort.
+        <br />
+        Existing users will remain in the experiment.
+      </ToggleField>
+
+      <FormGroup>
+        <Row>
+          <Col xs={24}>
+            <Branches />
+          </Col>
+        </Row>
+      </FormGroup>
+    </>
+  );
+}
+
+function Branches() {
+  const [branchesUpdated, setBranchesUpdated] = React.useState(Date.now());
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const { preferenceType } = data.arguments;
+
+  const handleClickAddBranch = () => {
+    const newBranch = { ratio: 1, slug: "", value: {}, groups: [] };
+    if (preferenceType === "boolean") {
```

Comment:
Probs a silly question, but is pref type supposed to be an editable field somewhere?

Reply:
oops! this should not be there at all. that's what i get for making a bowl of copy pasta.

Reply:
nice catch tho!